### ISSUE TITLE
WhitespaceAround ignores empty constructor whitespaces

### ISSUE THREAD

**Issue by pbaranchikov** [checkstyle/checkstyle#8237]
Check documentation: https://checkstyle.org/config_whitespace.html#WhitespaceAround
```
>javac Test.java
>cat Test.java 
package com.example;

public class Test {
    private Test(){} // Does not worn about absent space between ")" and "{"
}

>cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="WhitespaceAround">
            <property name="allowEmptyConstructors" value="true"/>
        </module>
    </module>
</module>
>java $RUN_LOCALE -jar checkstyle-8.32-all.jar -c config.xml Test.java
Starting audit...
Audit done
```
Expected: violation on the 4th line (constructor declaration). There should be space between `){`

Note, that this violation works Ok without `allowEmptyConstructors` option, but an extra violation is reported about empty constructors:
```
>cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="WhitespaceAround">
        </module>
    </module>
</module>

>java $RUN_LOCALE -jar checkstyle-8.32-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /Users/pavel/tmp/17/Test.java:4:19: '{' is not followed by whitespace. [WhitespaceAround]
[ERROR] /Users/pavel/tmp/17/Test.java:4:19: '{' is not preceded with whitespace. [WhitespaceAround]
[ERROR] /Users/pavel/tmp/17/Test.java:4:20: '}' is not preceded with whitespace. [WhitespaceAround]
Audit done.
Checkstyle ends with 3 errors.
```

I tested on Chekcstyle 8.32

**Comment by pbludov** [comment]
@pbaranchikov please update the issue description as described here https://checkstyle.org/report_issue.html#How_to_report_a_bug.3F

**Comment by pbaranchikov** [comment]
@pbludov done, thank you.

**Comment by pbludov** [comment]
Yes, this check works that way. There will be either none or three violations.

@romani @rnveach @strkkk should this check be changed to enforce a whitespace between `)` and `{`? Like this:
```java
public final class Util {
    private Util(){} // violation
    private Util(int a) {} // OK
}
```
The docs say that empty constructor, method, class, enum, interface, loop bodies

**Comment by pbaranchikov** [comment]
My personal point of view is that the main goal of 
> empty constructor, method, class, enum, interface, loop bodies may optionally be exempted from the policy

is to make it possible to create emty construcots/methods etc. with empty body without a space between curly branckets, i.e. `{}`

**Comment by romani** [comment]
> should this check be changed to enforce a whitespace between ) and {? 

Yes.
Issue is approved.

**Comment by Vyom-Yadav** [comment]
@romani I am on it.
